libmachine/provision/ubuntu.go

```diff
@@ -124,9 +124,22 @@ func (provisioner *UbuntuProvisioner) Hostname() (string, error) {
 
 func (provisioner *UbuntuProvisioner) SetHostname(hostname string) error {
 	cmd, err := provisioner.SSHCommand(fmt.Sprintf(
-		"sudo hostname %s && echo %q | sudo tee /etc/hostname && echo \"127.0.0.1 %s\" | sudo tee -a /etc/hosts",
+		"sudo hostname %s && echo %q | sudo tee /etc/hostname",
 		hostname,
 		hostname,
+	))
+
+	if err != nil {
+		return err
+	}
+
+	if err = cmd.Run(); err != nil {
```

Comment:
This should be a `:=` - otherwise you are setting the error from above (therefore it's not self-contained and anyone who changes the above code might have to change this too)